### PR TITLE
importer: validate serial volumes

### PR DESCRIPTION
**Pull request by kprzerwa**
* update invenio-app-ils

### CODE REVIEW

File: cds_ils/importer/documents/importer.py [hunk 1]
```diff
@@ -257,6 +257,29 @@ def _validate_match_identifiers(self, existing_document):
...
+        import_doc_serial = self.json_data.get("_serial", None)
+        existing_doc_serials = existing_document.relations.get('serial', [])
+        if import_doc_serial and existing_doc_serials:
+            import_volume = import_doc_serial.get("volume")
+            import_serial_title = import_doc_serial["title"].lower()
+
+            same_serial, same_volume, both_have_volumes = False, False, False
+
+            for serial in existing_doc_serials:
+                existing_volume = serial.get["volume"]
```
Comment (ntarocco) [inline cds_ils/importer/documents/importer.py:271]: It this a typo?
```suggestion
                existing_volume = serial.get("volume")
```

File: cds_ils/importer/documents/importer.py [hunk 2]
```diff
@@ -257,6 +257,29 @@ def _validate_match_identifiers(self, existing_document):
...
+                existing_title = serial["record_metadata"]["title"].lower()
+                if existing_title == import_serial_title:
+                    same_serial = True
+                    both_have_volumes = all([import_volume, existing_volume])
```
Comment (ntarocco) [inline cds_ils/importer/documents/importer.py:275]: Maybe the usage of `all` + `list` is a bit too much here?
```suggestion
                    both_have_volumes = import_volume and existing_volume
```
If not defined, they should be `None` I think.

File: cds_ils/importer/documents/importer.py [hunk 3]
```diff
@@ -257,6 +257,29 @@ def _validate_match_identifiers(self, existing_document):
...
+                    same_volume = existing_volume == import_volume
+            if same_serial and both_have_volumes and same_volume:
+                return True
+            return False
+        else:
+            return True
```
Comment (ntarocco) [inline cds_ils/importer/documents/importer.py:281]: Very minor and just IMHO, feel free to skip this.
I would move this `return True` more up to improve readability maybe? Even if it is considered not a very good practice to have multiple `return`...

```
if not (if import_doc_serial and existing_doc_serials):
    return True

....
```

File: cds_ils/importer/documents/importer.py [hunk 4]
```diff
@@ -284,9 +307,10 @@ def validate_found_matches(self, not_validated_matches):
                 import_doc_title.lower()
 
             valid_identifiers = self._validate_match_identifiers(document)
+            valid_serial_volumes = self._validate_volumes(document)
 
             if titles_not_equal or editions_not_equal or pub_year_not_equal \
-                    or not valid_identifiers:
+                    or not valid_identifiers or not valid_serial_volumes:
```
Comment (ntarocco) [inline cds_ils/importer/documents/importer.py:313]: I would twist the conditions to have `titles_equal and editions_equal...` to improve readability and avoid the negation...

File: cds_ils/importer/documents/importer.py (hunk 1 above)
Comment (kprzerwa) [reply cds_ils/importer/documents/importer.py:271]: it is! thank you!

**Pull request merged**